File: backend/src/main/java/dev/agiro/fanel/chores/api/ChoreDto.java
package dev.agiro.fanel.chores.api;

import java.time.Instant;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record ChoreDto(UUID id, UUID householdId, String title, UUID assigneeId, boolean done,
                       Instant createdAt, LocalDate dueDate, RecurrenceFrequency recurrenceFreq,
                       Integer recurrenceInterval, List<UUID> rotationMemberIds) {
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/ChoreDto.kt
package dev.agiro.fanel.android.data.remote

enum class RecurrenceFrequency {
    DAILY, WEEKLY, MONTHLY, YEARLY
}

data class ChoreDto(
    val id: String,
    val householdId: String,
    val title: String,
    val assigneeId: String?,
    val done: Boolean,
    val createdAt: String?,
    val dueDate: String?,
    val recurrenceFreq: RecurrenceFrequency?,
    val recurrenceInterval: Int?,
    val rotationMemberIds: List<String>?
)
